modules/core/src/lifecycle/create-props.ts

```diff
@@ -266,7 +266,9 @@ function getOwnProperty(object, prop) {
 
 function getComponentName(componentClass) {
   const componentName =
-    getOwnProperty(componentClass, 'layerName') || getOwnProperty(componentClass, 'componentName');
+    getOwnProperty(componentClass, 'layerName') ||
```

Comment:
Nit: This is exposing higher layer information into the lower layer (the point was that this file shouldn't know anything about layers)

Maybe no clean solution, like require those layers to define component name.

Reply:
Yes that could work. Not sure about whether there are any pitfalls of doing such things on the static objects, I have a sense that it ends up being a bit fiddly.

Reply:
At least according to this page `this` in static methods always refers to the class:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static#calling_static_members_from_another_static_method